src/components/UpupUploader/Preview.tsx

```diff
@@ -85,57 +76,15 @@ const Preview: FC<Props> = ({
-                                className="relative flex h-full w-full flex-col items-start dark:bg-[#1f1f1f] dark:text-[#fafafa]"
-                                onClick={() => {
-                                    if (onFileClick) onFileClick(file)
-                                }}
-                            >
-                                {(file as unknown as GoogleFile)
-                                    .thumbnailLink ||
-                                file.type.startsWith('image/') ? (
-                                    <img
-                                        src={
-                                            (file as unknown as GoogleFile)
-                                                .thumbnailLink ||
-                                            URL.createObjectURL(file)
-                                        }
-                                        alt=""
-                                        className={
-                                            'w-full rounded-md object-cover shadow ' +
-                                            (multiple ? 'h-40' : ' h-full')
-                                        }
-                                    />
-                                ) : (
-                                    <div
-                                        className={
-                                            'w-full rounded-md object-cover text-[#6b7280] shadow ' +
-                                            (multiple ? 'h-40' : 'h-[90%]')
-                                        }
-                                    >
-                                        <FileIcon name={file.name} />
-                                    </div>
-                                )}
-                                <div className="flex w-full items-center justify-between">
-                                    <div>
-                                        <p className="mt-1 text-xs font-medium">
-                                            {file.name}
-                                        </p>
-                                        <p className="text-[10px] font-medium text-gray-500">
-                                            {bytesToSize(file.size)}
-                                        </p>
-                                    </div>
-                                </div>
-                                <button
-                                    className="absolute -right-1 -top-1 rounded-full bg-black"
-                                    onClick={() => removeFile(i)}
-                                    type="button"
-                                >
-                                    <TbX className="h-4 w-4 text-white" />
-                                </button>
-                            </div>
+                        {files.map(file => (
+                            <PreviewComponent
+                                key={file.id || `${file.name}-${Date.now()}`}
```

Comment:
${file.name}-${Date.now()} can still result in duplicates if files have the same name and are added quickly.
maybe uuid is more suitable?

Reply:
@BSalaeddin Thanks man. Done